hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/SparkRDDWriteClient.java

```diff
@@ -380,11 +391,40 @@ protected void completeClustering(HoodieReplaceCommitMetadata metadata, JavaRDD<
   @Override
   protected HoodieTable<T, JavaRDD<HoodieRecord<T>>, JavaRDD<HoodieKey>, JavaRDD<WriteStatus>> getTableAndInitCtx(WriteOperationType operationType, String instantTime) {
     HoodieTableMetaClient metaClient = createMetaClient(true);
-    new SparkUpgradeDowngrade(metaClient, config, context).run(metaClient, HoodieTableVersion.current(), config, context, instantTime);
-    return getTableAndInitCtx(metaClient, operationType);
+    if (HoodieTableVersion.current() != metaClient.getTableConfig().getTableVersion()) {
```

Comment:
lets add a method `needsUpgradeOrDowngrade() ` to the upgradedowngrade class?

Reply:
Added